executor.go

```diff
@@ -71,13 +74,17 @@ func optExecutorInternalQueryClient(c InternalQueryClient) executorOption {
 func newExecutor(opts ...executorOption) *executor {
 	e := &executor{
 		client: newNopInternalQueryClient(),
+		work:   make(chan job, 2000),
```

Comment:
does the size of this buffer ever present a problem? i guess put another way, is there any downside to blocking the writes to this channel when the buffer is full? also, does changing the buffer size affect anything?

Reply:
I set it high for my testing to try to eliminate it as a variable. The more I think about it, the more I realize I don't know that much about channel performance characteristics.

There are going to be QUERY_CONCURRENCY goroutines putting things into this channel, and each one is going to put SHARDS_PER_NODE jobs into it per query in a tight loop. There are POOL_SIZE workers reading from it spending a widely variable amount of time on each job, but often ~1ms.

A large buffer means that the query handler goroutines which are populating the channel get to dump all their work in, and then wait on the result chan. A smaller channel means they might have to wait to dump their work into the work chan, but wait for less time on the result chan. So... setting this channel large is probably a waste of memory. Could it be totally unbuffered? Maybe... my instinct is it won't have much effect. I will test with setting it to POOL_SIZE, and unbuffered, and see what we get.